library/core/src/char/mod.rs

```diff
@@ -219,18 +218,24 @@ impl fmt::Display for EscapeUnicode {
 pub struct EscapeDefault(escape::EscapeIterInner<10>);
 
 impl EscapeDefault {
-    fn printable(chr: ascii::Char) -> Self {
-        let data = [chr];
-        Self(escape::EscapeIterInner::from_array(data))
+    #[inline]
+    const fn printable(c: ascii::Char) -> Self {
+        Self::ascii(c.to_u8())
```

Comment:
I think `ascii` is only used here? Then it doesn't make sense to define a separate function...
```suggestion
        Self(escape::EscapeIterInner::ascii(c.to_u8()))
```